frontend/videoCast/one_to_multiple_cast_skyway.py

```diff
@@ -91,53 +110,41 @@ async def handler(websocket, path):
-        elif msg_type == "connect_receiver":
-            print("connect_receiver")
-            if room["sender_socket"] is not None:
-                print("send")
-                if room["connect_num"] < MAX_CONNECT_NUM:
-                    promise = websocket.send(
-                        json.dumps(
-                            {
-                                "msg_type": "connect_receiver",
-                                "room_id": room_id,
-                                "skyway_room_id": room["skyway_room_id"],
-                                "peer_id": room["peer_id"],
-                            }
+                    else:
+                        # ルームの累積接続数が溢れそうだったら新しい部屋をsenderに作ってもらう
+                        promise = room["sender_socket"].send(
+                            json.dumps(
+                                {
+                                    "msg_type": "request_reconnect_sender",
+                                    "room_id": room_id,
+                                }
+                            )
                         )
-                    )
-
+                        promises.append(promise)
+                        async with lock:
+                            room["connect_num"] = 0
+            elif msg_type == "exit_room":
+                if websocket in room["connections"]:
                     async with lock:
-                        room["connect_num"] += 1
-                    promises.append(promise)
-                else:
-                    # ルームの累積接続数が溢れそうだったら新しい部屋をsenderに作ってもらう
-                    promise = room["sender_socket"].send(
-                        json.dumps(
-                            {"msg_type": "request_reconnect_sender", "room_id": room_id}
-                        )
-                    )
-                    promises.append(promise)
+                        room["connections"].remove(websocket)
+                if room["sender_socket"] is websocket:
                     async with lock:
+                        room["sender_socket"] = None
+                        room["peer_id"] = None
+                        room["skyway_room_id"] = None
                         room["connect_num"] = 0
-        elif msg_type == "exit_room":
-            if websocket in room["connections"]:
-                async with lock:
-                    room["connections"].remove(websocket)
-            if room["sender_socket"] is websocket:
-                async with lock:
-                    room["sender_socket"] = None
-                    room["peer_id"] = None
-                    room["skyway_room_id"] = None
-                    room["connect_num"] = 0
-        print("{}: {}".format(path, dictionary))
-        for p in promises:
-            try:
-                a = await p
-            except:
-                pass
+            print("{}: {}".format(path, dictionary))
+            for p in promises:
+                try:
+                    a = await p
```

Comment:
今回の変更には関係ないですが、この `a`に代入する作業って必要なんですかね？
```suggestion
                    await p
```
でも良さそうな気がしますがいかがなんでしょう。

Reply:
確かに `a` への代入は要らないと思います。

さらに言うとすべてのコルーチンの終了を待つには `for` 文ではなく次のように `asyncio.gather` の方が良いと思います。

```python
asyncio.gather(*promises, return_exceptions=True)
```

さらに言うと `promises` は次のように複数の接続に `send` する目的で使われていますが、それなら `send` ではなく[`websockets.broadcast`](https://websockets.readthedocs.io/en/stable/reference/utilities.html#websockets.broadcast) の方が良い気がします。

https://github.com/ueckoken/plarail2022/blob/9afa0a0ad907a8276caf55b8134b22f4606e6b3c/frontend/videoCast/one_to_multiple_cast_skyway.py#L82-L95

こうなると変更範囲がこのPRのnitsにしては大きすぎると思うのでissueを立てて別PRでやることにします。

Reply:
#141 を立てた